atom/browser/osr/osr_render_widget_host_view.cc

```diff
@@ -909,128 +859,70 @@ void OffScreenRenderWidgetHostView::SetNeedsBeginFrames(
   begin_frame_timer_->SetActive(needs_begin_frames);
 
   if (software_output_device_) {
-    software_output_device_->SetActive(needs_begin_frames && painting_, false);
-  }
-}
-
-void CopyBitmapTo(const SkBitmap& destination,
-                  const SkBitmap& source,
-                  const gfx::Rect& pos) {
-  char* src = static_cast<char*>(source.getPixels());
-  char* dest = static_cast<char*>(destination.getPixels());
-  int pixelsize = source.bytesPerPixel();
-
-  int width =
-      pos.x() + pos.width() <= destination.width()
-          ? pos.width()
-          : pos.width() - ((pos.x() + pos.width()) - destination.width());
-  int height =
-      pos.y() + pos.height() <= destination.height()
-          ? pos.height()
-          : pos.height() - ((pos.y() + pos.height()) - destination.height());
-
-  if (width > 0 && height > 0) {
-    for (int i = 0; i < height; i++) {
-      memcpy(dest + ((pos.y() + i) * destination.width() + pos.x()) * pixelsize,
-             src + (i * source.width()) * pixelsize, width * pixelsize);
-    }
+    software_output_device_->SetActive(painting_, false);
   }
-
-  destination.notifyPixelsChanged();
 }
 
 void OffScreenRenderWidgetHostView::OnPaint(const gfx::Rect& damage_rect,
                                             const SkBitmap& bitmap) {
   TRACE_EVENT0("electron", "OffScreenRenderWidgetHostView::OnPaint");
 
-  HoldResize();
```

Comment:
How come we don't need a resize lock any more?

Reply:
The resize lock interferes in a breaking way with the scale factor change, the view gets stuck waiting for a frame size to release the lock that never actually arrives. I couldn't find a good solution to this problem, so I decided to remove the lock, since `ResizeLock` is going away (replaced by surface synchronization if I'm correct) in the near future.
I could try to pour some more work into trying to fix this instead of removing it, probably gonna do that.